core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks.go

```diff
@@ -304,7 +286,10 @@ func (am *AssetManagement) IsFungibleAssetLocked(stub shim.ChaincodeStubInterfac
         log.Error(errorMsg)
         return false, errors.New(errorMsg)
     }
-    iccResp := stub.InvokeChaincode(am.interopChaincodeId, [][]byte{[]byte("IsFungibleAssetLocked"), []byte(assetAgreement.Type), []byte(strconv.FormatInt(int64(assetAgreement.NumUnits), 10)), []byte(assetAgreement.Recipient), []byte(assetAgreement.Locker)}, "")
+
+    contractId := generateSHA256HashInBase64Form(assetAgreement.Type + ":" + strconv.Itoa(int(assetAgreement.NumUnits)) +
+			":" + assetAgreement.Locker + ":" + assetAgreement.Recipient)
```

Comment:
I don't think this is correct. The idea here was to have the `LockFungibleAsset` function return a unique contract Id that the caller could store and then use in functions like `IsFungibleAssetLocked`, `ClaimFungibleAsset`, and `UnlockFungibleAsset`. The above contract Id generation will produce collisions across multiple locks over similar quantities of an asset made by a given party for another.

Reply:
Made changes to address this behavior.